misaligned.cpp

```diff
@@ -1,21 +1,70 @@
 #include <assert.h>
 #include <iostream>
+#include <sstream>
+#include <string>
+#include <algorithm>
+#include <tuple>
 
-int printColorMap() {
+int printColorMap(std::ostream& out) {
     const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
     const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
     int i = 0, j = 0;
     for (i = 0; i < 5; i++) {
         for (j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+            out << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
         }
     }
     return i * j;
 }
 
+// Test - core part
+std::tuple<int, int, int, int> testPrintColorMap() {
+    std::stringstream ss;
+    printColorMap(ss);
+    // test
+    const char* majorColor[] = { "White", "Red", "Black", "Yellow", "Violet" };
+    const char* minorColor[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
+    int nCntNumMatch = 0, nCntMajorMatch = 0, nCntMinorMatch = 0, nCntFullMatch = 0;
+    std::string inputStr;
+    int i = 0, j = 0;

```

Comment:
Naming: `i` is the majorColorIndex and `j` is the minorColorIndex. We can call them as such :)

Reply:
Done